### ISSUE TITLE
Refuse to work unless admin acknowledges perf warnings

### ISSUE THREAD

**Issue by pulsejet** [pulsejet/memories#646]
Similar to Nextcloud's unsupported browser message.

Other warnings to add:
1/ Not installed preview generator app.
2/ Imagick extension

Also show warning on main page if nothing is indexed yet (also add this to FAQ)

**Comment by szaimen** [comment]
> 2/ Imagick extension

Only if imaginary is not configured?

**Comment by pulsejet** [comment]
> Only if imaginary is not configured?

There's a direct dependency on Imagick currently [here](https://github.com/pulsejet/memories/blob/79347da0d02981b1427591aef3d5a82306c0fff8/lib/Controller/ImageController.php#L320-L323) and [here](https://github.com/pulsejet/memories/blob/79347da0d02981b1427591aef3d5a82306c0fff8/lib/Service/FileRobotMagick.php#L143). Migrating the HEIC -->JPEG conversion to imaginary is probably trivial but the editor will probably take a lot more time to support multiple backends (assuming Imaginary even provides such an API).

**Comment by szaimen** [comment]
Ah so use imagick directly? Interesting!

Imaginary supports these: https://github.com/nextcloud/server/blob/9d2d3d482bce004d134dd87d7b37c3dfb6513412/lib/private/Preview/Imaginary.php#L60

**Comment by pulsejet** [comment]
Yup, getting the `decodable` method to work with both should be easy. For the `FileRobotMagick` class, it does seem like Imaginary has a image processing [API](https://github.com/h2non/imaginary#supported-image-operations), but that'll need some better abstractions (an adapter?) to prevent huge code duplication.

**Comment by vitis586** [comment]
Can one of the perf warnings be the use of external storage with transcoding?
I found a workaround myself but it was more elegant without it. 🙂
